src/tools/rbd/action/Config.cc

```diff
@@ -75,6 +90,186 @@ std::ostream& operator<<(std::ostream& os,
+      "\"who\": \"client.*\", " // XXXMG: use the current user id instead of *?
+      "\"key\": \"" + key + "\""
+    "}";
+  bufferlist in_bl;
+  bufferlist out_bl;
+  r = rados.mon_command(cmd, in_bl, &out_bl, nullptr);
+  if (r < 0) {
+    std::cerr << "rbd: error reading " << key << ": " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  std::cout << out_bl.c_str();
+  return 0;
+}
+
+void get_global_set_arguments(po::options_description *positional,
+                            po::options_description *options) {
+  add_key_option(positional);
+  positional->add_options()
+    ("value", "config value");
+}
+
+int execute_global_set(const po::variables_map &vm,
+                     const std::vector<std::string> &ceph_global_init_args) {
+  librados::Rados rados;
+  int r = utils::init_rados(&rados);
+  if (r < 0) {
+    return r;
+  }
+
+  std::string key;
+  r = get_key(vm, 0, &rados, &key);
+  if (r < 0) {
+    return r;
+  }
+
+  std::string value = utils::get_positional_argument(vm, 1);
+  std::string cmd =
+    "{"
+      "\"prefix\": \"config set\", "
+      "\"who\": \"client\", " // XXXMG: use the current user id instead of *?
+      "\"name\": \"" + key + "\", "
+      "\"value\": \"" + stringify(json_stream_escaper(value)) + "\""
+    "}";
+  bufferlist in_bl;
+  bufferlist out_bl;
+  r = rados.mon_command(cmd, in_bl, &out_bl, nullptr);
+  if (r < 0) {
+    std::cerr << "rbd: error setting " << key << ": " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  return 0;
+}
+
+void get_global_remove_arguments(po::options_description *positional,
+                                 po::options_description *options) {
+  add_key_option(positional);
+}
+
+int execute_global_remove(
+    const po::variables_map &vm,
+    const std::vector<std::string> &ceph_global_init_args) {
+  librados::Rados rados;
+  int r = utils::init_rados(&rados);
+  if (r < 0) {
+    return r;
+  }
+
+  std::string key;
+  r = get_key(vm, 0, &rados, &key);
+  if (r < 0) {
+    return r;
+  }
+
+  std::string cmd =
+    "{"
+      "\"prefix\": \"config rm\", "
+      "\"who\": \"client\", " // XXXMG: use the current user id instead of *?
+      "\"name\": \"" + key + "\""
+    "}";
+  bufferlist in_bl;
+  bufferlist out_bl;
+  r = rados.mon_command(cmd, in_bl, &out_bl, nullptr);
+  if (r < 0) {
+    std::cerr << "rbd: error removing " << key << ": " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  return 0;
+}
+
+void get_global_list_arguments(po::options_description *positional,
+                             po::options_description *options) {
+  at::add_format_options(options);
+}
+
+int execute_global_list(const po::variables_map &vm,
+                      const std::vector<std::string> &ceph_global_init_args) {
+  at::Format::Formatter f;
+  int r = utils::get_formatter(vm, &f);
+  if (r < 0) {
+    return r;
+  }
+
+  MonClient mon_client(g_ceph_context);
```

Comment:
I think it would be nicer to just send a "config dump" mon command instead of bootstrapping up a second MonClient.